### DISCUSSION TITLE
Make WorkBench status message less ambigous

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5652] [Feature Requests]
Regarding WorkBench status message when you click on Validate/Upload/Unupload:

> As to the "Beginning the Validating..." message, currently, it can mean at least 3 things and there is no way for the front end to differentiate between them:
> The upload worker is preparing to validate the DS and validation should begin soon
> The upload worker is currently busy with another DS and validation would not begin until that DS is validated
> The upload worker is down or not responding. The validation would not begin and the "Beginning the Validating" would stay up forever
> All of this also applies to Upload and Rollback.

Maybe front-end can say something after 30seconds, or maybe a way to detect if the worker process is busy should be added

**Comment by maxpatiiuk** [comment]
From @grantfitzsimmons:

> Currently the WorkBench just hangs on the validation or upload function if the [`worker`](https://github.com/specify/specify7#the-specify-7-worker) process is not running.
> 
> It would be nice to have some icon and tooltip (or something similar) that displays a warning if it pings the server and the worker is not running. Even if it is busy, it is helpful to have some indication.
> 
> There have been many instances where users ask [support@specifysoftware.org](mailto:support@specifysoftware.org) for help due to this process not running. Without an indication for a user, it seems that the WorkBench is just always busy, even when it is definitely not the case.

**Comment by grantfitzsimmons** [comment]
Importantly, there should be a way to warn the user that the worker is not running when when using the WorkBench in a visual way! Most important part of my issue 😉
